### ISSUE TITLE
I haven't been able to run the demo successfully yet.

### ISSUE THREAD

**Issue by dackless** [OGRECave/ogre-next#462]
请考虑重新设计资源管理方案
我认为目前的方案有以下问题：
1.过于依赖搜索目录。随便一个Demo就要设置好几个搜索目录。而代码里面仅仅只写一个文件名 。读代码的时候，并不知道这个文件在哪里。还需要去目录里找一遍。
2.不可以引用相对目录的资源，而需要再次走搜索目录里查找文件。比如材质A引用了B文件，B文件还需去搜索目录里找，而不是在当前目录里找
3.隐藏了一些自动操作。比如自动加载材质。我认为可能c++用户更喜欢自己手动管理
总之，我感觉不太好定制，而且对新手不太友好。
当然我也没有深入使用，因为我至今没有在仓库外成功运行demo（资源这块把我难住了）。

Translated From Google:

Please consider redesigning the resource management solution
I think the current solution has the following problems:
1. Too dependent on search directories. For any demo, you need to set up several search directories. And there is only one file name in the code. When reading the code, you don’t know where the file is. You need to search the directory again.
2. You cannot reference resources in relative directories, but need to search the file in the search directory again. For example, material A references file B, and file B needs to be found in the search directory instead of the current directory
3. Some automatic operations are hidden. For example, automatically load materials. I think C++ users may prefer to manage them manually
In short, I feel that it is not easy to customize and not very friendly to novices.
Of course, I haven’t used it in depth, because I have not successfully run the demo outside the ogre-next repo so far (the resources management have stumped me).
